tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -4114,3 +4115,278 @@ def test_16_barbican(self):
                     reporting.set_test_script_status(tvaultconf.FAIL)
                     reporting.add_test_script(test[0])
                     reporting.test_case_to_write()
+
+    # Test case automated #OS-2031
+    # http://192.168.15.51/testlink/linkto.php?tprojectPrefix=OS&item=testcase&id=OS-2031
+    @decorators.attr(type='workloadmgr_api')
+    def test_17_barbican(self):
+        test_var = "tempest.api.workloadmgr.barbican.rbd_device_cleanup_.attach_ceph_volume_"
+        restore_tests = [[test_var + "selectiverestore_api", "selective"], [test_var + "inplacerestore_api", "inplace"], [test_var + "oneclickrestore_api", "oneclick"]]
+
+        for restore_test in restore_tests:
+            try:
+                reporting.add_test_script(restore_test[0])
+
+                try:
+                    order_uuid = self.create_secret_order("sec_order1")
+                    LOG.debug("Created secret order: {}".format(order_uuid))
+                    time.sleep(10)
+                    secret_uuid = self.get_secret_from_order(order_uuid)
+                    LOG.debug("Getting secret key from secret order: {}".format(secret_uuid))
+                    reporting.add_test_step("Creating secret key using secret order", tvaultconf.PASS)
+                except Exception as e:
+                    LOG.error(f"Exception: {e}")
+                    raise Exception("Getting secret key from secret order")
+
+                vm_id = self.create_vm(vm_cleanup=True)
+                # Get the details of luks-ceph (encrypted ceph) volumes from setup
+                volume_type_id = ""
+                for vol in CONF.volume.volume_types:
+                    if (vol.lower().find("luks") != -1):
+                        if (vol.lower().find("ceph") != -1) or (vol.lower().find("tripleo") != -1) or (
+                                vol.lower().find("rbd") != -1):
+                            volume_type_id = CONF.volume.volume_types[vol]
+                            LOG.debug("Encrypted volume for ceph/rbd found : {}".format(volume_type_id))
+
+                if not volume_type_id:
+                    raise Exception("Exiting the test as encrypted ceph/rbd volume type is not found")
+
+                vol_count = 2
+                volumes = []
+                disk_names = ["vda", "vdb", "vdc"]
+                for i in range(vol_count):
+                    volume_id = self.create_volume(
+                        volume_type_id=CONF.volume.volume_types['tripleo'])
```

Comment:
If volume type of tripleo does not exist on the setup, this will throw error.

Reply:
made change